Crash Report

```~~~ 1618954637488.stacktrace ~~~

java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:523)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1101)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
	... 1 more
Caused by: java.io.FileNotFoundException: No content provider: tag:calculator2.android.com,2021-04-20:819293632
	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1684)
	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1514)
	at android.content.ContentResolver.openInputStream(ContentResolver.java:1198)
	at dev.patrickgold.florisboard.ime.clip.provider.FileStorage.cloneURI(FileStorage.kt:42)
	at dev.patrickgold.florisboard.ime.clip.provider.FlorisContentProvider.insert(FlorisContentProvider.kt:77)
	at android.content.ContentProvider$Transport.insert(ContentProvider.java:325)
	at android.content.ContentResolver.insert(ContentResolver.java:1844)
	at dev.patrickgold.florisboard.ime.clip.provider.ClipboardItem$Companion.fromClipData(FlorisDatabase.kt:123)
	at dev.patrickgold.florisboard.ime.clip.FlorisClipboardManager.onPrimaryClipChanged(FlorisClipboardManager.kt:214)
	at android.content.ClipboardManager.reportPrimaryClipChanged(ClipboardManager.java:270)
	at android.content.ClipboardManager$1.lambda$dispatchPrimaryClipChanged$0$ClipboardManager$1(ClipboardManager.java:64)
	at android.content.-$$Lambda$ClipboardManager$1$hQk8olbGAgUi4WWNG4ZuDZsM39s.run(Unknown Source:2)
	at android.os.Handler.handleCallback(Handler.java:888)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:213)
	at android.app.ActivityThread.main(ActivityThread.java:8178)
	... 3 more
```

--- a/app/src/main/java/dev/patrickgold/florisboard/ime/clip/provider/FlorisDatabase.kt
+++ b/app/src/main/java/dev/patrickgold/florisboard/ime/clip/provider/FlorisDatabase.kt
@@ -5,6 +5,7 @@ import android.content.ContentValues
 import android.net.Uri
 import android.provider.BaseColumns
 import androidx.room.*
+import dev.patrickgold.florisboard.ime.clip.FlorisClipboardManager
 import dev.patrickgold.florisboard.ime.core.FlorisBoard
 import java.io.Closeable
 
@@ -103,12 +104,10 @@ data class ClipboardItem(
          * @param cloneUri Whether to store the image using [FlorisContentProvider].
          */
         fun fromClipData(data: ClipData, cloneUri: Boolean) : ClipboardItem {
-
             val type = when {
-                data.getItemAt(0)?.uri != null -> ItemType.IMAGE
-                data.getItemAt(0)?.text != null -> ItemType.TEXT
-                else -> null
-            }!!
+                data.getItemAt(0)?.uri != null && data.description.hasMimeType("image/*") -> ItemType.IMAGE
+                else -> ItemType.TEXT
+            }
 
             val uri = if (type == ItemType.IMAGE) {
                     if (data.getItemAt(0).uri.authority == FlorisContentProvider.CONTENT_URI.authority || !cloneUri){
@@ -122,7 +121,7 @@ data class ClipboardItem(
                     }
             } else { null }
 
-            val text = data.getItemAt(0).text?.toString()
+            val text = FlorisBoard.getInstanceOrNull()?.let { data.getItemAt(0).coerceToText(it).toString() } ?: "#ERROR"
             val mimeTypes = when (type) {
                 ItemType.IMAGE -> {
                     (0 until data.description.mimeTypeCount).map {

